lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -72,12 +82,24 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
       |> String.replace(" ", "")
       |> Integer.parse()
 
-    %{
-      ops
-      | unspent_outputs: [
-          %UnspentOutput{from: address, amount: initial_supply * @unit_uco, type: {:NFT, address}}
-        ]
-    }
+    # internal check before minting
+    case initial_supply <= @nft_id_limit_in_digits && initial_supply >= 1 do
```

Comment:
Also this kind of verification could be done before, in the pending_transaction.ex but the token standard will make changes, so to avoid changes for nothing, we will handle later